### PR TITLE
Updated to support GreyNoise SDK

### PR DESCRIPTION
**Pull request by superducktoes**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- ✅ Testing of all the changes has been performed (for bug fixes / features)
- ✅ The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- ✅ Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- ✅ Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- ✅ Verify all checks are passing.
- ✅ Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- ✅ Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- ✅ Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- ✅ Other (please describe): Modified to use the GreyNoise Python SDK. 

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
Updated the app to use the latest GreyNoise SDK, as well as relevant data paths. 

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.
No functionality of the app has changed. Can review on a call if necessary. 

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by superducktoes** [comment]
I have most of the linting issues resolved but just curious about the error from the last linting test Imports are incorrectly sorted and/or formatted. What is the preferred method for sorting the imports?

**Comment by bradchiappetta** [comment]
@arathore-crest : how can we get this PR reviewed and approved to update the integration?  We'd like to get this update out to our users?

**Comment by arathore-crest** [comment]
@bradchiappetta, Before submitting this app for review, we'll need to perform some basic developer checks and QA sanity testing. We'll pick up this app based on our backlog.

Thanks,
Aarav

**Comment by bradchiappetta** [comment]
@arathore-crest : would you be able to at least provide an estimated time frame for when someone is going to be able to review?  We'd like to be able to set appropriate expectations internally.  Thanks for understanding.

File: greynoise_connector.py [hunk 1]
```diff
@@ -32,6 +43,7 @@ def __init__(self):
         self._session = None
         self._app_version = None
         self._api_key = None
+        self._integration_name = "splunk-soar-v2.5.0"
```
Comment (bradchiappetta) [inline greynoise_connector.py:46]: @superducktoes : looks like the desire here is for this to be v2.2.0.  We should update that, and also this value should probably be in the const file.  Can you please push an update?

Comment (superducktoes) [reply greynoise_connector.py]: @bradchiappetta just updated

File: readme.html [hunk 2]
```diff
@@ -67,12 +76,32 @@ <h4>on poll</h4>
...
     <h4>test connectivity</h4>
     <p>Test connectivity to GreyNoise. Requires a valid paid or free community API key.</p>
-  
+
     <h2>Legal</h2>
     <p>For terms and legal information, please visit <a target="_blank" rel="noopener noreferrer" href="https://greynoise.io/terms">https://greynoise.io/terms</a></p>
-  
+    <h2>Port Information</h2>
+      <p>
+          The app uses HTTP/ HTTPS protocol for communicating with the ThreatQuotient server. Below are the default ports used by the Splunk SOAR Connector.
```
[the remark this answers is not in the document]
Comment (gusdadiya-crest) [reply readme.html:87]: Yes, It is a standard template for port details that we add in SOAR Connectors app. I mistakenly forget to change the app name in that. I have modified it now.

File: greynoise.json [hunk 3]
```diff
@@ -101,7 +101,7 @@
                     "data_type": "string",
                     "example_values": [
                         "The IP has never been observed scanning the Internet",
-                        "The IP has been observed by the GreyNoise sensor network"
+                        "The IP has been observed by the test sensor network"
```
Comment (bradchiappetta) [inline greynoise.json:104]: @dhwanis-crest : this change should be reverted.  This should read "GreyNoise sensor network"

File: greynoise.json [hunk 4]
```diff
@@ -590,7 +596,7 @@
...
```
Comment (bradchiappetta) [inline greynoise.json:599]: @dhwanis-crest : please revert as well.

File: greynoise.json [hunk 5]
```diff
@@ -705,23 +732,26 @@
                 {
                     "data_path": "action_result.data.*.link",
                     "data_type": "string",
+                    "contains": [
+                        "url"
+                    ],
                     "example_values": [
-                        "https://viz.greynoise.io/riot/1.2.3.4"
+                        "https://viz.testurl.io/riot/1.2.3.4"
```
Comment (bradchiappetta) [inline greynoise.json:739]: @dhwanis-crest : why are we removing / changing these sample values?  This seems like an odd pattern?

File: greynoise.json (hunk 3 above)
Comment (bradchiappetta) [reply greynoise.json:104]: Yes, we would prefer that we keep these as we've noted them.  Thank you.

File: greynoise.json (hunk 5 above)
Comment (dhwanis-crest) [reply greynoise.json]: Done

File: greynoise.json (hunk 4 above)
Comment (dhwanis-crest) [reply greynoise.json]: Done

File: greynoise.json (hunk 3 above)
Comment (dhwanis-crest) [reply greynoise.json]: Done

**Comment by mattsayar-splunk** [comment]
Hi @bradchiappetta sorry about the confusion about this app's availability on Splunkbase. Since moving SOAR apps to Splunkbase, we discovered we can't call this app "Greynoise" because there's already a Splunk app called Greynoise. We're working on the backend to make it work nicely, but a temporary solution could be to rename this app to something else. Feel free to reach out to me msayar@splunk.com

File: greynoise_connector.py [hunk 6]
```diff
@@ -182,92 +245,86 @@ def _check_apikey(self, action_result):
...
-        return action_result.set_status(phantom.APP_SUCCESS)
+        license_type = "enterprise"
+        message = "Enterprise license."
 
-    def _lookup_ip(self, param):
-        action_result = self.add_action_result(ActionResult(dict(param)))
-        ret_val = self._check_apikey(action_result)
-        if phantom.is_fail(ret_val):
-            return ret_val
+        if(config.get("license_type") == "community"):
```
Comment (rgil-splunk) [inline greynoise_connector.py:284]:
```suggestion
        if config.get("license_type") == "community":
```

same for any other cases with parentheses around `if`/`while` statements

File: greynoise_connector.py [hunk 7]
```diff
@@ -313,273 +412,203 @@ def _community_lookup_ip(self, param):
             if result_data["riot"] is False and result_data['noise'] is False:
```
Comment (rgil-splunk) [inline greynoise_connector.py:412]:
```suggestion
            if not result_data["riot"] and not result_data['noise']:
```

File: greynoise_connector.py [hunk 8]
```diff
@@ -605,21 +634,48 @@ def _on_poll(self, param):
...
+            return action_result.set_status(phantom.APP_SUCCESS, "No Data Found")
 
-        if phantom.is_fail(ret_val):
-            return action_result.set_status(
-                phantom.APP_ERROR, "Failed to process the query"
-            )
-        else:
-            return action_result.set_status(phantom.APP_SUCCESS)
+        self.save_progress("Creating containers")
+        for i in query_results[0][-int(param["size"]):]:
```
Comment (rgil-splunk) [inline greynoise_connector.py:645]: `i` isn't an index here (which is what `i`,`j`,`k`, etc. is conventionally used for), can we name this something more descriptive or at least `result`?

**Review by rgil-splunk — APPROVED** [review]

File: greynoise_connector.py (hunk 8 above)
Comment (dhwanis-crest) [reply greynoise_connector.py]: Done

File: greynoise_connector.py (hunk 7 above)
Comment (dhwanis-crest) [reply greynoise_connector.py]: Done

File: greynoise_connector.py (hunk 6 above)
Comment (dhwanis-crest) [reply greynoise_connector.py]: Done

**Comment by dhwanis-crest** [comment]
Hi @bradchiappetta ,
Should we proceed with merging this MR or do you want to change the app name as per the suggestion?
Please let us know.

**Comment by bradchiappetta** [comment]
@dhwanis-crest : we are speaking with @mattsayar-splunk on Monday to discuss what needs to be done here.  Please hold on merging until that discussion occurs.  Appreciate your assistance and getting this approved.

**Comment by bradchiappetta** [comment]
@dhwanis-crest : we've pushed our last update, this should be ready to review and merge after your approval.

**Review by ricardo-phantom — APPROVED** [review]

**Pull request merged**